jscomp/runtime/js_string.ml

```diff
@@ -106,326 +101,363 @@ external get : t -> int -> t = ""
 ]}
-
 *)
 
-(** [codePointAt n s] returns the code point at position [n] within string [s] as a [Some] value. The return value handles code points greater than or equal to [0x10000]. If there is no code point at the given position, the function returns [None].
+external codePointAt : t -> pos:int -> int option = "codePointAt"
+[@@mel.send]
+(** [codePointAt s ~pos] returns the code point at position [pos] within string
+    [s] as a [Some] value. The return value handles code points greater than or
+    equal to [0x10000]. If there is no code point at the given position, the
+    function returns [None].
 
 {[
-  codePointAt 1 {js|¿😺?|js} = Some 0x1f63a
-  codePointAt 5 "abc" = None
+  codePointAt {js|¿😺?|js} 1 = Some 0x1f63a
+  codePointAt "abc" 5 = None
 ]}
 *)
-external codePointAt : int -> int option = "codePointAt"
-[@@mel.send.pipe: t]
+
 (** ES2015 *)
 
-external concat : t -> t = "concat"
-[@@mel.send.pipe: t]
-(** [concat append original] returns a new string with [append] added after [original].
+external concat : t -> other:t -> t = "concat"
+[@@mel.send]
+(** [concat original ~other] returns a new string with [other] added
+    after [original].
 
 {[
-  concat "bell" "cow" = "cowbell";;
+  concat "cow" ~other:"bell" = "cowbell";;
 ]}
 *)
 
-external concatMany : t array -> t = "concat"
-[@@mel.send.pipe: t] [@@mel.splice]
-(** [concat arr original] returns a new string consisting of each item of an array of strings added to the [original] string.
+external concatMany : t -> strings:t array -> t = "concat"
+[@@mel.send] [@@mel.splice]
+(** [concatMany original ~strings] returns a new string consisting of each item
+    of the array of strings [strings] added to the [original] string.
 
 {[
-  concatMany [|"2nd"; "3rd"; "4th"|] "1st" = "1st2nd3rd4th";;
+  concatMany "1st" ~strings:[|"2nd"; "3rd"; "4th"|] = "1st2nd3rd4th";;
 ]}
 *)
 
-external endsWith : t -> bool = "endsWith"
-[@@mel.send.pipe: t]
-(** ES2015:
-    [endsWith substr str] returns [true] if the [str] ends with [substr], [false] otherwise.
+external endsWith : t -> suffix:t -> bool = "endsWith"
+[@@mel.send]
+(** [endsWith str ~suffix] returns [true] if the [str] ends with [suffix],
+    [false] otherwise.
 
 {[
-  endsWith "World!" "Hello, World!" = true;;
-  endsWith "world!" "Hello, World!" = false;; (* case-sensitive *)
-  endsWith "World" "Hello, World!" = false;; (* exact match *)
+  endsWith "Hello, World!" ~suffix:"World!" = true;;
+  endsWith "Hello, World!" ~suffix:"world!" = false;; (* case-sensitive *)
+  endsWith "Hello, World!" ~suffix:"World" = false;; (* exact match *)
 ]}
 *)
 
-(** [endsWithFrom ending len str] returns [true] if the first [len] characters of [str] end with [ending], [false] otherwise. If [n] is greater than or equal to the length of [str], then it works like [endsWith]. (Honestly, this should have been named [endsWithAt], but oh well.)
+external endsWithFrom : t -> suffix:t -> len:int -> bool = "endsWith"
```

Comment:
Like in the Array PR, these functions can be unified, so there's a single `endsWith` with optional labelled args that match [MDN spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith):

```ocaml
external endsWith : u -> ?searchString:u -> ?endPosition:int -> unit -> bool
  = "endsWith"
[@@mel.send]
```